testing_guide.md

```diff
@@ -0,0 +1,698 @@
+
+```shell
+bitcoind -i2pacceptincoming=1
+```
+
+Next, run an I2P node that is not accepting incoming connections and confirm that **your I2P address is different for each outgoing connection**.
+
+```shell
+bitcoind -i2pacceptincoming=0
+```
+
+Once the test is successful, don't forget to [stop your node and clean the datadir](#4-reset-testing-environment).
+
+## Migrating Legacy Wallets to Descriptor Wallets
+
+Descriptor wallets [were introduced with the Bitcoin Core 0.21 release](https://achow101.com/2020/10/0.21-wallets) and became the default wallet type with the Bitcoin Core 23.0 release. Descriptor wallets are ultimately a better design, that's why the [legacy wallet type and BDB format will soon be unsupported](https://github.com/bitcoin/bitcoin/issues/20160), which creates the need for a migration mechanism.
+
+> :flashlight: Want to dive deeper into the changes? Check the [PR Review Club Meeting for PR#19602](https://bitcoincore.reviews/19602)
+
+We are going to test this migration mechanism by using the new `migratewallet` RPC to migrate our legacy wallets. There is an extremely large number of possible configurations for the scripts that Legacy wallets can know about, be watching for, and be able to sign for. This guide cannot cover them all. Therefore **testing configurations other than the testing scenarios of this guide is encouraged and could provide valuable feedback to strengthen this migration mechanism**.
+
+We will test the `migratewallet` RPC by creating a new legacy wallet and then migrating it. It is not required to have funds in it as the migration deals with keys and scripts, not really transactions. Even without funds, you can check that the keys and scripts were migrated correctly.
+
+### What to expect after the migration
+
+A new descriptor wallet creates 8 descriptors by default, 2 (external, internal) for each supported script type:
+
+- P2PKH: `pkh([<fingerprint>/44'/1'/0']<xpub>/{0,1}/*)#<checksum>`
+- P2SH(P2WPKH): `sh(wpkh([<fingerprint>/49'/1'/0']<xpub>/{0,1}/*))#<checksum>`
+- P2WPKH: `wpkh([<fingerprint>/84'/1'/0']<xpub>/{0,1}/*)#<checksum>`
+- P2TR: `tr([<fingerprint>/86'/1'/0']<xpub>/{0,1}/*)#<checksum>`
+
+> :information_source: You can verify that with the `listdescriptors` RPC on a new descriptor wallet. For more information on output descriptors, see [doc/descriptors.md][descriptors.md].
+
+A legacy wallet with an HD chain has an HD seed. This seed is hashed according to the BIP32 specification to become the BIP32 master key which everything else is then derived from. We need to associate those addresses with a descriptor. Because of that, in addition to the above, the new migrant wallet is expected to have:
+
+- A range combo descriptor for the external addresses: `combo(<xpub>/0'/0'/*')#<checksum>`
+- A range combo descriptor for the internal addresses: `combo(<xpub>/0'/1'/*')#<checksum>`
+- A combo descriptor for the hd seed: `combo(<hdseed>)#<checksum>`
```

Comment:
I believe this should be two combo descriptors for the hd seed, right? One for internal (change) and one for external (receive)? I ran through this test scenario [twice](https://github.com/satsie/bitcoin-notez/blob/master/v24.0rc1-testing.md#migrating-legacy-wallets-to-descriptor-wallets) and both times I got two combo descriptors for the hd seed.

Reply:
I've read Will's answer on IRC and I thought that you cleared this out, just repeating it here to make sure. 

The testing guide says:
> For wallets containing non-HD keys, each key will have its own combo descriptor.

That's the [non-hd keys section](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/24.0-Release-Candidate-Testing-Guide/#non-hd-keys). There, we are using the `importprivkey` RPC. That's where that second combo descriptor comes from, it's not internal/external. One is from the hd seed and one is for the non-hd key. 

I've also have a note at the what to expect section that says
> ℹ️ The HD seed is a valid key that could receive Bitcoin to it even though its corresponding addresses would never be given out.

Essentially that means that the HD seed is another non-hd key.

Please let me know if how it is stated in the testing guide creates confusion.

Reply:
ah!! I understand now :) Thanks for clarifying. I totally forgot the test has you import a non-HD key. It was a lot of steps for someone like me that has zero idea what they are doing haha :sweat_smile:. It makes sense that this would be the reason for the second combo descriptor. In general I'm not used to working with wallets that have non-HD keys imported in, so it was a new paradigm for me to wrap my head around.

I think at the end of the migration, when I scrolled back up to the "What to expect after the migration" section I assumed this part was going to be a checklist:

- A range combo descriptor for the external addresses: `combo(<xpub>/0'/0'/*')#<checksum>`
-  A range combo descriptor for the internal addresses: `combo(<xpub>/0'/1'/*')#<checksum>`
-  A combo descriptor for the hd seed: `combo(<hdseed>)#<checksum>`

And for the most part, it is. I just got so hung up on the last bit where I had two combo descriptors, I didn't try to read more into the stuff immediately below about the non-HD keys. 

Reflecting on it now, the wording of:

> For wallets containing non-HD keys, the new migrant wallet is expected... 

probably also threw me off. I followed all the steps in the test guide, so the part about importing the non-HD key didn't seem optional to me, yet this statement felt like I only needed to be concerned if I did something advanced or special.

Anyways, all of this is to say I think only extremely new readers will find themselves confused the way I did, and in general I'd say it's a non-issue. Certainly not worth splitting hairs over!

This concludes my feedback on the guide. Thank you for all the work you put into developing it, hosting PR review club, and sticking around to answer questions. As a beginner it was so wonderful to have a guide that walked me through all the steps, and taught me about different facets of the system I knew nothing about. Glad I was able to do some testing on the upcoming release, and that wouldn't have been possible for me without this guide! :star2: